File: 3314. construct-the-minimum-bitwise-array-i.py
from typing import List


class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        res = []
        for n in nums:
            if n & 1:
                res.append(n & ~(((n + 1) & ~n) >> 1))
            else:
                res.append(-1)
        return res


# Test cases
if __name__ == "__main__":
    solution = Solution()
    
    # Test case 1
    nums1 = [2,3,5,7]
    output1 = solution.minBitwiseArray(nums1)
    print(f"Test 1: nums = {nums1}")
    print(f"Output: {output1}, Expected: [-1,1,4,3]")
    
    # Test case 2
    nums2 = [11,13,31]
    output2 = solution.minBitwiseArray(nums2)
    print(f"Test 2: nums = {nums2}")
    print(f"Output: {output2}, Expected: [9,12,15]")
    
    print("\nTest cases executed!")

File: 3314. construct-the-minimum-bitwise-array-i.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public int[] minBitwiseArray(List<Integer> nums) {
        int[] res = new int[nums.size()];
        
        for (int i = 0; i < nums.size(); i++) {
            int n = nums.get(i);
            if ((n & 1) == 1) {
                res[i] = n & ~(((n + 1) & ~n) >> 1);
            } else {
                res[i] = -1;
            }
        }
        
        return res;
    }

    // Test cases
    public static void main(String[] args) {
        Solution solution = new Solution();
        
        // Test case 1
        List<Integer> nums1 = Arrays.asList(2, 3, 5, 7);
        int[] output1 = solution.minBitwiseArray(nums1);
        System.out.println("Test 1: nums = [2,3,5,7]");
        System.out.println("Output: " + Arrays.toString(output1) + ", Expected: [-1,1,4,3]");
        
        // Test case 2
        List<Integer> nums2 = Arrays.asList(11, 13, 31);
        int[] output2 = solution.minBitwiseArray(nums2);
        System.out.println("Test 2: nums = [11,13,31]");
        System.out.println("Output: " + Arrays.toString(output2) + ", Expected: [9,12,15]");
        
        System.out.println("\nTest cases executed!");
    }
}
